### PR TITLE
[feat](stats) Support to kill analyze

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

1. Report error if submit analyze jobs when stats table is not available
2. Support kill analyze
3. Support  cancel sync analyze 

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by englefly** [comment]
I have following concerns
1. to cancel a analyze job, we mark the job as killed, but do not clear its partial results. is that right?
2. could you add a regression test to demonstrate KILL ANALYZE works?

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -4753,6 +4755,13 @@ kill_stmt ::=
     :}
     ;
 
+kill_analysis_job_stmt ::=
+    KW_KILL KW_ANALYZE INTEGER_LITERAL:value
```
[the remark this answers is not in the document]
Comment (Kikyou1997) [reply fe/fe-core/src/main/cup/sql_parser.cup:4759]: If you submit async analyze, FE would return a job id

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 2]
```diff
@@ -76,12 +79,14 @@ public class AnalysisManager {
     // The time field that needs to be displayed
     private static final String LAST_EXEC_TIME_IN_MS = "last_exec_time_in_ms";
 
-    private final ConcurrentMap<Long, Map<Long, AnalysisTaskInfo>> analysisJobIdToTaskMap;
+    private final ConcurrentMap<Long, Map<Long, BaseAnalysisTask>> analysisJobIdToTaskMap;
```
[the remark this answers is not in the document]
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:82]: It means task id

**Review by englefly — APPROVED** [review]

**Comment by Kikyou1997** [comment]
> I have following concerns
> 
> 1. to cancel a analyze job, we mark the job as killed, but do not clear its partial results. is that right?
> 2. could you add a regression test to demonstrate KILL ANALYZE works?

1. Column level stats if is final wroten, so it's ok to leave some partition level stats in table, and it won't cause loading incomplete stats
2. Yes, we should, however, the pipeline is parallel running and it's hard to have a determined  job's state for it

**Pull request merged**